core/src/main/java/org/stellar/anchor/sep10/Sep10Service.java

```diff
@@ -288,6 +288,8 @@ void validateChallengeRequest(
     Set<Sep10Challenge.Signer> signers = fetchSigners(account);
     // the signatures must be greater than the medium threshold of the account.
     int threshold = account.getThresholds().getMedThreshold();
+    Network network = new Network(appConfig.getStellarNetworkPassphrase());
+    String homeDomain = extractHomeDomainFromChallengeXdr(request.getTransaction(), network);
```

Comment:
This method can throw `InvalidSep10ChallengeException` which is an SDK-specific error, we should catch it and throw an AP error here.